web/src/main/java/org/cbioportal/web/SampleController.java

```diff
@@ -75,14 +90,43 @@
         @RequestParam(defaultValue = "ASC") Direction direction
     ) {
         String sort = sortBy == null ? null : sortBy.getOriginalValue();
-        
+        List<String> studyIds = null;
+        if (usingAuth()) {
+            /*
+             If using auth, filter the list of samples returned using the list of study ids the
+             user has access to. If the user has access to no studies, the endpoint should not 403,
+             but instead return an empty list.
+            */
+            studyIds = studyService
+                .getAllStudies(
+                    null,
+                    projection.name(),
+                    PagingConstants.MAX_PAGE_SIZE,
+                    0,
+                    null,
+                    direction.name()
+                )
+                .stream()
+                .map(CancerStudy::getCancerStudyIdentifier)
+                .collect(Collectors.toList());
+        }
+
         if (projection == Projection.META) {
             HttpHeaders httpHeaders = new HttpHeaders();
-            httpHeaders.add(HeaderKeyConstants.TOTAL_COUNT, sampleService.getMetaSamples(keyword).getTotalCount().toString());
+            if (studyIds != null && studyIds.isEmpty()) {
```

Comment:
I do not like this solution, and I am open to suggestions. The problem I was having was, when the user has permission to access no studies, the search uses an empty list. Depending on my implementation, the SQL query was either malformed, or returned all matching studies for the keyword. What I need it to do is return no samples when the list of study ids is empty, but not null.

There _has_ to be a cleaner way to do this in `SampleMapper.xml`. Help?